service/history/historyEngine.go

```diff
@@ -499,16 +474,6 @@ func (e *historyEngineImpl) createMutableState(
 			e.logger,
 			namespaceEntry,
 		)
-	} else if namespaceEntry.IsGlobalNamespace() {
```

Comment:
you should also delete the following lines:
465
466
469
477-484

basically since the NDC will be default enabled, the mutable state initialization logic should be
```go
		newMutableState = newMutableStateBuilderWithVersionHistories(
			e.shard,
			e.shard.GetEventsCache(),
			e.logger,
			namespaceEntry,
		)
```